api/src/auth/drivers/oauth2.ts

```diff
@@ -97,7 +98,9 @@ export class OAuth2AuthDriver extends LocalAuthDriver {
 		} catch (e) {
 			throw handleError(e);
 		}
-
+		if (validatePayload(this.config.publicRegistrationFilter, userInfo).length > 0) {
+			throw new InvalidCredentialsException();
+		}
```

Comment:
Lets add an extra check that checks if `this.config.publicRegistrationFilter` is configured, or have it default to `{}` if not. Lets also add a `logger.warn` (or info? 🤔 ) when throwing the error, so admins have a way to debug why they're not allowed to login 👍🏻